docs/cuda-manual-windows-build.md

```diff
@@ -0,0 +1,153 @@
+
+- The installers for CUDA Toolkit and MSVC do not put their respective binaries
+  in the user's `$PATH` environment variable.
+- A default installation of MSYS2 ignores the global and user's `$PATH` anyway.
+- Having files in `c:\Program Files\` (path with a space) causes all sorts of
+  issues in all layer of the compilation stack.  In particular, [cgo directives
+  do not work](https://github.com/golang/go/issues/45637) with such paths.
+- An [outdated CUDA package](https://github.com/barnex/cuda5) currently used by
+  `gominer` has wrong cgo include paths in any case.
+- CUDA requires `cl.exe` (MSVC compiler) to build its binary kernels, but Go
+  does not accept it as a possible `cgo` compiler.
+
+
+All of these must be resolved in other to correctly build the CUDA version of
+`gominer` on Windows.  The automatic builder script uses a few tricks and some
+reasonable assumptions to attempt to solve building for the most common install
+scenarios.  But for the cases where that is not sufficient, the following
+document lists a general procedure for setting the required Windows environment.
+
+## Install the Required Tools
+
+- Download and install the appropriate NVIDIA driver
+  - https://www.nvidia.com/download/index.aspx
+- Download and install the NVIDIA CUDA Toolkit:
+  - https://developer.nvidia.com/cuda-toolkit
+- Download and install Microsoft Visual Studio:
+  - https://visualstudio.microsoft.com/vs/community/
+  - Ensure the "Desktop Development with C++" component will be installed
+
+Restart the computer and ensure Windows can recognize the NVIDIA GPU as such.
+
+## Setup MSYS2
+
+While it _may_ be possible to build on PowerShell or even `cmd.exe`, going
+through MSYS2 means the changes are more explicit and generally don't require
+clicking through GUIs to find settings.
+
+- Download and install [MSYS2](https://www.msys2.org/)
+  - Make sure you uncheck `Run MSYS2 now.`
+- Launch the `MSYS2 MINGW64` shell from the start menu
+  - NOTE: The `MSYS2` installer will launch the `UCRT64` shell by default if
+    you didn't uncheck `Run MSYS2 now` as instructed.  That shell will not work,
+    so close it if you forgot to uncheck it in the installer.
+- From within the `MSYS2 MINGW64` shell enter the following commands to install
+  `gcc`, `git`, `go`, `unzip`:
+  - `pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-tools mingw-w64-x86_64-go git unzip`
+  - `git clone https://github.com/decred/gominer`
+
+
+## Create junction to NVIDIA Toolkit
+
+Now comes the tricky bit. `cgo` flags don't currently support paths with spaces
+in them. Therefore, we'll create a junction to the NVIDIA Toolkit install path
+in order to be able to specify it.
+
+- Determine the NVIDIA Toolkit install path.
+  - This is generally `c:\Program Files\NVIDIA GPU Toolkit\CUDA\vXX.Y`
+- Create a junction for the target directory from within MSYS2:
+  - `cd gominer`
+  - `cmd.exe //c mklink /J nvidia "c:\Program Files\NVIDIA GPU Toolkit\CUDA\vXX.Y"`
```

Comment:
I think it has to be a double slash on the J so the msys2 cmd translation layer doesn't eat it.

```suggestion
  - `cmd.exe //c mklink //J nvidia "c:\Program Files\NVIDIA GPU Toolkit\CUDA\vXX.Y"`
```